aws-educate-tw/aws-educate-tpet-backend
# Scrum 157 list webhooks

## Description

完成GET /webhooks 這隻api的邏輯

## Type of Change

- [ ] 🐛 Bug Fix（修復 bug）
- [ ] 💄 UI/UX（介面或使用者體驗優化）
- [ ] 🔨 Refactor（程式碼重構）
- [ ] 📝 Documentation（文件更新）
- [ ] 🔧 Configuration（配置變更）
- [ ] 🚀 Performance（效能優化）
- [ ] ✅ Test（測試相關）
- [ ] 🔒 Security（安全性相關）

## Related Issues

[SCRUM-157](https://aws-educate-cloud-ambassador-dev-team.atlassian.net/browse/SCRUM-157)

## Testing

### How to test
1. Make sure you are in the Postman `Harry-dev` environment
2. Call the `POST /auth/login` API ([Postman link](https://aws-educate-tw.postman.co/workspace/TPET-API-Development-Workspace~b6915213-64a3-408d-b297-45f472fa0a6b/request/36968695-bb6e4623-89eb-4787-9ab0-9744f2f1b207?action=share&creator=31171060&ctx=documentation&active-environment=31171060-a68faedf-68d1-4502-9ef2-3dd830e580eb)) to get a valid JWT token in the **dev** environment.
3. Cal the `GET /webhooks` API ([Postman link](https://aws-educate-tw.postman.co/workspace/TPET-API-Development-Workspace~b6915213-64a3-408d-b297-45f472fa0a6b/request/31171060-e562e3f7-1b69-459e-9849-5c23d8a6beb0?action=share&creator=31171060&ctx=documentation&active-environment=31171060-a68faedf-68d1-4502-9ef2-3dd830e580eb)) to get a list of webhooks


## Screenshots/Videos
If the settings are correct, you will get a response like this.
```
{
    "data": [
        {
            "bcc": [
                "harryup2000@gmail.com"
            ],
            "subject": "test",
            "cc": [
                "harryup2000@gmail.com"
            ],
            "webhook_name": "test",
            "created_at": "2024-12-16T06:36:40Z",
            "sender_local_part": "aws",
            "attachment_file_ids": [
                "0a688qwdqwdqwdqwdqwdwqe3e"
            ],
            "is_generate_certificate": false,
            "display_name": "aws",
            "sequence_number": 1.0,
            "surveycake_link": "https://www.surveycake.com/s/D6R2R",
            "iv_key": "sdfsdfsdfsdfs",
            "hash_key": "sfdsfsdfsdfsd",
            "webhook_url": "https://api.tpet.aws-educate.tw/dev/trigger-webhook/ef9qwdqwdqwdqwdqwdqwdqwdqwqcb1",
            "template_file_id": "0aqwdqwdqwdqwdqwdqwdf5e3e",
            "reply_to": "harryup2000@gmail.com",
            "webhook_type": "surveycake",
            "webhook_id": "eqwdqwdqwdqwdqwdqwdwefcb1"
        },
        {
            "bcc": [
                "harryup2000@gmail.com"
            ],
            "subject": "aws",
            "cc": [],
            "webhook_name": "aws",
            "created_at": "2024-12-16T06:45:14Z",
            "sender_local_part": "aws",
            "attachment_file_ids": [
                "362krsjvbweifnwoifwpeofweff66"
            ],
            "is_generate_certificate": false,
            "display_name": "aws",
            "sequence_number": 2.0,
            "surveycake_link": "https://example.com",
            "iv_key": "sdfsdfsdfsdfs",
            "hash_key": "sfdsfsdfsdfsd",
            "webhook_url": "https://api.tpet.aws-educate.tw/dev/trigger-webhook/6weoifhweofihwkejfnwkejfnkwjefnwefweede",
            "template_file_id": "750weoijfwoeifweifnwoeifowef733b4cb5",
            "reply_to": "harryup2000@gmail.com",
            "webhook_type": "surveycake",
            "webhook_id": "61wefwefwekcjnwoeincwoefweweede"
        },
        {
            "bcc": [
                "harryup2000@gmail.com"
            ],
            "subject": "aws",
            "cc": [
                "harryup2000@gmail.com"
            ],
            "webhook_name": "aws",
            "created_at": "2024-12-16T06:46:55Z",
            "sender_local_part": "aws",
            "attachment_file_ids": [],
            "is_generate_certificate": false,
            "display_name": "aws",
            "sequence_number": 3.0,
            "surveycake_link": "https://example.com",
            "iv_key": "sdfsdfsdfsdfs",
            "hash_key": "sfdsfsdfsdfsd",
            "webhook_url": "https://api.tpet.aws-educate.tw/dev/trigger-webhook/fweoifowiejfwoeijfwoeocijspoasdjwedwec",
            "template_file_id": "750oicnsodicnoeiwf0w9eufwoeifwe5",
            "reply_to": "harryup2000@gmail.com",
            "webhook_type": "surveycake",
            "webhook_id": "fweuifnwieufw9euhfikjsndksjfsdfewwec"
        }
    ],
    "page": 1,
    "limit": 10,
    "total_count": 15,
    "sort_order": "ASC"
}
```


## Checklist

- [x] 我已經測試過這些變更
- [ ] 我已經更新相關文件
- [x] 我的程式碼遵循專案的程式碼規範
- [x] 我已經處理所有的 console warnings/errors
- [x] 我已經移除所有的 console.log
- [ ] 我已經新增必要的測試案例
- [x] 所有現有的測試都能通過

## Additional Notes

N/A


[SCRUM-157]: https://aws-educate-cloud-ambassador-dev-team.atlassian.net/browse/SCRUM-157?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
